### DISCUSSION TITLE
Dynamic system in EnergyPlus?

### DISCUSSION THREAD

**Discussion by OOMAR77** [NREL/EnergyPlus#9749] [Q&A]
I want to simulate an office room using EnergyPlus where the exterior wall contains cylinders in the middle. These cylinders are formed from different materials and at a Time t=t1, the cylinders should rotate. Is it possible to build such a design using Energyplus or Dhoul I search for another software

**Comment by mjwitte** [comment]
I can't think of a way to model this in EnergyPlus.

**Comment by Myoldmopar** [comment]
I agree with @mjwitte  that there's nothing like this in Energyplus itself.  

There might be a fun workaround.  Make that surface that has the cylinders to be adiabatic and set the radiative properties so that it doesn't really interact thermally with the zone at all.  And then create a plain plug load (ElectricEquipment) object in the zone.  Then use a Python Plugin to do the detailed modeling of the cylinders and update the equipment power schedule accordingly so that the overall impact on the zone is similar to what the wall is doing.  In the Python plugin, you'd just need to use the outdoor temp and solar gain on the wall as sensors, do your calculations, and actuate a schedule.  Might not be too hard to prototype out...
